--- ctdb-2.0/config/ctdb.service.0007~	2012-12-12 20:35:37.164884611 +0100
+++ ctdb-2.0/config/ctdb.service	2012-12-12 20:35:37.164884611 +0100
@@ -0,0 +1,19 @@
+[Unit]
+Description=CTDB
+After=network.target
+
+[Service]
+Type=forking
+LimitCORE=infinity
+PIDFile=/run/ctdbd/ctdbd.pid
+ExecStartPre=/etc/ctdb/systemd/ctdb_drop_all_public_ips
+ExecStartPre=/etc/ctdb/systemd/ctdb_check_persistent_databases.pre
+ExecStart=/etc/ctdb/systemd/ctdb.systemd
+ExecStartPost=/etc/ctdb/systemd/ctdb_set_ctdb_variables.post
+ExecStop=/usr/bin/ctdb shutdown
+ExecStopPost=-/etc/ctdb/systemd/ctdb_drop_all_public_ips
+KillMode=control-group
+Restart=no
+
+[Install]
+WantedBy=multi-user.target
--- ctdb-2.0/config/systemd/ctdb_check_persistent_databases.pre.0007~	2012-12-12 20:35:37.164884611 +0100
+++ ctdb-2.0/config/systemd/ctdb_check_persistent_databases.pre	2012-12-12 20:35:37.164884611 +0100
@@ -0,0 +1,20 @@
+#!/bin/sh
+#
+
+# Avoid using root's TMPDIR
+unset TMPDIR
+
+[ -z "$CTDB_BASE" ] && {
+    export CTDB_BASE="/etc/ctdb"
+}
+
+[ -z "$CTDB_VARDIR" ] && {
+    export CTDB_VARDIR="/var/ctdb"
+}
+
+. $CTDB_BASE/functions
+. $CTDB_BASE/init_functions
+
+. /etc/sysconfig/ctdb
+
+check_persistent_databases || exit 1
--- ctdb-2.0/config/systemd/ctdb_drop_all_public_ips.0007~	2012-12-12 20:35:37.164884611 +0100
+++ ctdb-2.0/config/systemd/ctdb_drop_all_public_ips	2012-12-12 20:35:37.164884611 +0100
@@ -0,0 +1,22 @@
+#!/bin/sh
+#
+
+# Avoid using root's TMPDIR
+unset TMPDIR
+
+[ -z "$CTDB_BASE" ] && {
+    export CTDB_BASE="/etc/ctdb"
+}
+
+[ -z "$CTDB_VARDIR" ] && {
+    export CTDB_VARDIR="/var/ctdb"
+}
+
+. $CTDB_BASE/functions
+. $CTDB_BASE/init_functions
+
+. /etc/sysconfig/ctdb
+
+drop_all_public_ips
+
+exit 0
--- ctdb-2.0/config/systemd/ctdb_set_ctdb_variables.post.0007~	2012-12-12 20:35:37.164884611 +0100
+++ ctdb-2.0/config/systemd/ctdb_set_ctdb_variables.post	2012-12-12 20:35:37.164884611 +0100
@@ -0,0 +1,26 @@
+#!/bin/sh
+#
+
+# Avoid using root's TMPDIR
+unset TMPDIR
+
+[ -z "$CTDB_BASE" ] && {
+    export CTDB_BASE="/etc/ctdb"
+}
+
+[ -z "$CTDB_VARDIR" ] && {
+    export CTDB_VARDIR="/var/ctdb"
+}
+
+. $CTDB_BASE/functions
+. $CTDB_BASE/init_functions
+
+. /etc/sysconfig/ctdb
+
+if wait_until_ready ; then
+    set_ctdb_variables
+else
+    exit 1
+fi
+
+exit 0
--- ctdb-2.0/config/systemd/ctdb.systemd.0007~	2012-12-12 20:35:37.164884611 +0100
+++ ctdb-2.0/config/systemd/ctdb.systemd	2012-12-12 20:35:37.164884611 +0100
@@ -0,0 +1,32 @@
+#!/bin/sh
+#
+
+# Avoid using root's TMPDIR
+unset TMPDIR
+
+[ -z "$CTDB_BASE" ] && {
+    export CTDB_BASE="/etc/ctdb"
+}
+
+[ -z "$CTDB_VARDIR" ] && {
+    export CTDB_VARDIR="/var/ctdb"
+}
+
+. $CTDB_BASE/functions
+. $CTDB_BASE/init_functions
+
+. /etc/sysconfig/ctdb
+
+ctdbd=${CTDBD:-/usr/sbin/ctdbd}
+
+if [ "$CTDB_SUPPRESS_COREFILE" = "yes" ]; then
+    ulimit -c 0
+else
+    ulimit -c unlimited
+fi
+
+build_ctdb_options
+
+exec $ctdbd "$CTDB_OPTIONS" >/dev/null 2>&1 </dev/null
+
+exit 1
